### PR TITLE
Make alike Discord

### PR DESCRIPTION
**Pull request by Monochromish**
This PR:
* Adds animation to the guild icon, just like in Discord.
* Made a few changes to the color pallet.
* Removed unnecessary borders after each message in chat log, instead added a darker background effect upon hover.
* Made Guild Name Bold.
* Made changes to timestamp to make it alike Discord.
* Added a divider between guild information and chat log + Few more changes…
* Bumped version from `2.4.2` to `2.4.3`

Screenshots:
* Before
image
* After
En845nRqQA

Thank you!

### CODE REVIEW

**Review by ItzDerock — APPROVED** [review]
Everything looks great!
I'll pull this in a bit and give it a few tests and merge if everything goes well.

**Comment by Monochromish** [comment]
hey @ItzDerock, just added support for hyperlinks! Please review it as well 😄

image

Thank you!

File: src/exporthtml.ts [hunk 1]
```diff
@@ -126,23 +126,32 @@ function generateTranscript<T extends ReturnTypes>(messages: discord.Message[],
...
-            messageContentContentMarkdownSpan.innerHTML = formatContent(
-                message.content,
-                channel,
-                message.webhookId !== null
-            );
-
-            messageContentContentMarkdown.appendChild(messageContentContentMarkdownSpan);
-            messageContentContent.appendChild(messageContentContentMarkdown);
-            messageContent.appendChild(messageContentContent);
+            if (message.content.startsWith('http://') || message.content.startsWith('https://')) {
```
Comment (ItzDerock) [inline src/exporthtml.ts:129]: I would use a URL regex instead of checking to see if the content starts with http(s)://

**Review by ItzDerock — CHANGES_REQUESTED** [review]

File: src/exporthtml.ts (hunk 1 above)
Comment (Monochromish) [reply src/exporthtml.ts:129]: On it.

Comment (Monochromish) [reply src/exporthtml.ts]: Fixed.

File: src/exporthtml.ts [hunk 2]
```diff
@@ -552,4 +552,14 @@ function formatBytes(bytes: number, decimals = 2) {
...
 }
 
+function validateURL(url: string) {
+    var regex = /[-a-zA-Z0-9@:%._\+~#=]{1,256}\.[a-zA-Z0-9()]{1,6}\b([-a-zA-Z0-9()@:%_\+.~#?&//=]*)?/gi;
+
+    if (!regex.test(url)) {
+        return false;
+    } else {
+        return true;
+    }
```
Comment (ItzDerock) [inline src/exporthtml.ts:562]: This function can be simplified to just

```js
function validateURL(url: string) {
    return !(/[-a-zA-Z0-9@:%._\+~#=]{1,256}\.[a-zA-Z0-9()]{1,6}\b([-a-zA-Z0-9()@:%_\+.~#?&//=]*)?/gi.test(url));
}
```

File: src/exporthtml.ts [hunk 3]
```diff
@@ -126,7 +126,7 @@ function generateTranscript<T extends ReturnTypes>(messages: discord.Message[],
 
         // message content
         if(message.content) {
-            if (message.content.startsWith('http://') || message.content.startsWith('https://')) {
+            if (validateURL(message.content) == true) {
```
Comment (ItzDerock) [inline src/exporthtml.ts:129]: the `== true` here is not required

**Review by ItzDerock — CHANGES_REQUESTED** [review]

File: src/exporthtml.ts (hunk 2 above)
Comment (Monochromish) [reply src/exporthtml.ts:562]: Working on both the requests.

Comment (Monochromish) [reply src/exporthtml.ts]: Resolved.

File: src/exporthtml.ts (hunk 3 above)
Comment (Monochromish) [reply src/exporthtml.ts]: Resolved.

**Review by ItzDerock — APPROVED** [review]

**Comment by ItzDerock** [comment]
giving it a test now

**Comment by Monochromish** [comment]
I am going to make a few more changes, will push in ∼30 mins ish.
Will ping you here once I have them tested and such'.

**Comment by ItzDerock** [comment]
There are also a few typescript issues you need to resolve
image

**Comment by Monochromish** [comment]
Working on better reply css.

**Comment by Monochromish** [comment]
@ItzDerock Please review it again. Changed a bunch of stuff as well and fixed the previous typescript bugs.

**Comment by Monochromish** [comment]
Current Preview

**Comment by Skippydingledoo** [comment]
:o

**Review by ItzDerock — APPROVED** [review]
looks good, lmk if there are any other changes you want to make before I merge

**Comment by Monochromish** [comment]
You can merge this one, I'll make another PR later.

**Pull request merged**